test/Quake/merge_rotation.qke

```diff
@@ -45,14 +47,16 @@ func.func @duplicate_rotation_check2() {
 }
 
 // CHECK-LABEL:   func.func @duplicate_rotation_check2() {
-// CHECK-DAG:       %[[VAL_0:.*]] = arith.constant 2.300000e-01 : f64
-// CHECK-DAG:       %[[VAL_1:.*]] = arith.constant 5.900000e-01 : f64
-// CHECK-DAG:       %[[VAL_2:.*]] = quake.alloca !quake.veq<2>
-// CHECK:           %[[VAL_3:.*]] = quake.extract_ref %[[VAL_2]][0] : (!quake.veq<2>) -> !quake.ref
-// CHECK:           quake.rx (%[[VAL_1]]) %[[VAL_3]] : (f64, !quake.ref) -> ()
-// CHECK:           quake.rx (%[[VAL_1]]) %[[VAL_3]] : (f64, !quake.ref) -> ()
-// CHECK:           quake.rx (%[[VAL_0]]) %[[VAL_3]] : (f64, !quake.ref) -> ()
-// CHECK:           %[[VAL_4:.*]] = quake.mz %[[VAL_3]] : (!quake.ref) -> i1
+// CHECK-DAG:       %[[VAL_0:.*]] = arith.constant 2.30{{.*}} : f64
+// CHECK-DAG:       %[[VAL_1:.*]] = arith.constant 5.90{{.*}} : f64
```

Comment:
I think we should keep the `e-01` as part of check. Otherwise we risk being off by an order of magnitude.

Reply:
Checking floating point values verbatim is notoriously perilous. I'm trying to straddle the line between not bothering to check the value at all and not hard-coding in potentially brittle and arbitrary formatting.

I think if floating point is suddenly off by order(s) of magnitude, there will be some major upstream problems in MLIR. Hopefully, no one does that, but if they do, they are going to be other unhappy customers.